_pytest/recwarn.py

```diff
@@ -110,15 +110,10 @@ def warns(expected_warning, *args, **kwargs):
             return func(*args[1:], **kwargs)
 
 
-class RecordedWarning(object):
-    def __init__(self, message, category, filename, lineno, file, line):
-        self.message = message
-        self.category = category
-        self.filename = filename
-        self.lineno = lineno
-        self.file = file
-        self.line = line
-
+RecordedWarning = namedtuple('RecordedWarning', (
```

Comment:
If you feel this way, perhaps you should just add a `__repr__` method then? This would solve the issue and could go into `master`.